frontend/src/Judge/components/VideoSubmission.js

```diff
@@ -8,25 +8,68 @@ const YOUTUBE_BASE_URL = 'https://www.youtube.com/watch?v='
 const VIMEO_BASE_URL = 'https://vimeo.com/'
 
 const PlayerContainer = styled.div`
-  /* Center the top-most 'div' that ReactPlayer wraps around the 'iframe'  */
-  & > div {
-    margin: 0 auto;
-  }
+  height: 100%;
+  display: inline-block;
 `
 
-const VideoSubmission = props => (
-  <PlayerContainer>
-    {props.provider === 'youtube' ? (
-      <ReactPlayer url={`${YOUTUBE_BASE_URL}${props.videoId}`} playing controls />
-    ) : props.provider === 'vimeo' ? (
-      <ReactPlayer url={`${VIMEO_BASE_URL}${props.videoId}`} playing controls />
-    ) : null}
-  </PlayerContainer>
-)
-
-VideoSubmission.propTypes = {
-  provider: PropTypes.string.isRequired,
-  videoId: PropTypes.string.isRequired
-}
+class VideoSubmission extends Component {
+  static propTypes = {
+    provider: PropTypes.string.isRequired,
+    videoId: PropTypes.string.isRequired
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      height: 0,
+      width: 0
+    }
+    this.updateDimensions = this.updateDimensions.bind(this)
+  }
+
+  updateDimensions () {
```

Comment:
If you change this to `updateDimensions = () => {`, you can remove `this.updateDimensions = this.updateDimensions.bind(this)` above.